packages/frontpage/lib/api/post.ts

```diff
@@ -0,0 +1,84 @@
+import "server-only";
+import * as db from "../data/db/post";
+import * as atproto from "../data/atproto/post";
+import { ensureUser, getBlueskyProfile } from "../data/user";
+import { DataLayerError } from "../data/error";
+import { sendDiscordMessage } from "../discord";
+import { invariant } from "../utils";
+import { TID } from "@atproto/common-web";
+
+export type ApiCreatePostInput = {
+  title: string;
+  url: string;
+  createdAt: Date;
```

Comment:
I think our AppView API shouldn't accept a `createdAt` field. Client can specify this for backfill purposes but there's no need for them to be doing this via the AppView, they should be writing archives to the repo directly. This brings consistency to createComment also.